docs/applications.md

```diff
@@ -92,3 +146,131 @@ is the volume from list.
 To attach volume you can run `attach <volume name> <app name> [mount point]`. Where `<volume name>`
 is the volume from list, `<app name>` - name of application you want to attach the volume, `[mount point]` - the
 mount point of volume attached to the app (may be omitted).
+
+Notice: if you are on QEMU there is a limited number of exposed ports.
+Add some if you want to expose more.
+
+```console
+hostfwd:
+         {{ .DefaultSSHPort }}: 22
+         5912: 5901
+         5911: 5900
+         8027: 8027
+         8028: 8028
+```
+
+## Application Deployment Details
+
+EVE can load and run application images from different sources. In addition,
+eden deploys its own http server and docker registry, to simplify presenting
+local files and container images to a running EVE device.
+
+Each source has its default format, but you can override it with
+the `--format` flag (`container`,`qcow2` or `raw`). The defaults are:
+
+* `docker://` - OCI image
+* `http://` - VM qcow2
+* `https://` - VM qcow2 image
+* `file://` - VM qcow2 image
```

Comment:
Can you please add "from local file".

Reply:
Sure, I added information to the whole section.